ghost/mailgun-client/lib/mailgun-client.js

```diff
@@ -107,13 +110,19 @@ module.exports = class MailgunClient {
         }
     }
 
+    /**
+     * Fetches events from Mailgun
+     * @param {Object} mailgunOptions
+     * @param {Function} batchHandler
+     * @param {Object} options
+     * @param {Number} options.maxEvents Not a strict maximum. We stop fetching after we reached the maximum AND received at least one event after begin (not equal) to prevent deadlocks.
+     * @returns {Promise<void>}
```

Comment:
The result has been removed because it didn't make much sense as we could also access the results by a batchHandler implementation. That reduces the need to concat the array on every loop iteration. It is also a lot more flexible to work with inside the batchHandler.